File: OnlineGravitryp/WebContent/resources/Ship.js
/**
 * 
 */
function p (centre, shape, xwidth, ywidth, shipdir, velocity) {
	
	this.maxspd = 5.0;
	this.rotspd = 4.0;
	this.accel = 0.1;
	this.sprite = new Sprite(centre, shape, xwidth, ywidth, shipdir);
	this.velocity = velocity;
	
	this.rotateLeft = function() {
		this.direction.rotate(-this.rotspd);
		this.sprite.hitbox(sprite.centre, sprite.direction);
	}
	
	this.rotateRight = function() {
		this.direction.rotate(this.rotspd);
		this.sprite.hitbox(sprite.centre, sprite.direction);
	}
	
	this.setVelocity = function(v) {
		if(p.length() > this.maxspd) {
			v = new Point(v.x * (this.maxspd/v.length()), v.y * (this.maxspd/v.length()));
		}
		this.velocity = v;
	}
	
	this.move = function() {
		var update = new Point(sprite.direction.x, sprite.direction.y);
		update.scale(this.accel);
		this.setVelocity(new Point(this.velocity.x + update.x, this.velocity.y + update.y));
	}
	
	this.slow = function() {
		if(this.velocity > this.accel) {
			this.velocity.scale(1 - this.accel);
		}
		else {
			this.velocity = new Point(0, 0);
		}
	}
	
	this.applyGravity = function(planets) {
		var g_xsum = 0;
		var g_ysum = 0;
		
		for(var i = 0; i < planets.arguments; i++) {
			var dx = planets[i].sprite.centre.x - this.centre.x;
			var dy = planets[i].sprite.centre.y - this.centre.y;
			var distsq = dx*dx + dy*dy;
			g_xsum += planets.gc * (dx / distsq);
			g_ysum += planets.gc * (dy / distsq);
		}
		
		this.setVelocity(new Point(this.velocity.x + g_xsum, this.velocity.y + g_ysum));
	}
}

File: Gravitryp v2/src/objects/Ship.java
package objects;

public class Ship extends Sprite {

	private Point velocity; //vector with a maximum length, points where ship is going and also signifies speed
	private double rotspd = 4.0; //rotation speed of ship. (could become a settable)
	private double maxspd = 5.0; //maximum speed of ship. (could become a settable)
	private double accel = 0.1; //acceleration of ship. (could become a settable)
	
	public Ship(Point centre, String shape, double xwidth, double ywidth, Point shipdir, Point velocity) {
		super(centre, shape, xwidth, ywidth, shipdir);
		this.setVelocity(velocity);
	}
	
	//rotates the ship to the left
	public void rotateLeft() {
		this.getDirection().rotate(-rotspd);
		this.setHitbox(this.getCentre(), this.getDirection());
	}
	
	//rotates the ship to the right
	public void rotateRight() {
		this.getDirection().rotate(rotspd);
		this.setHitbox(this.getCentre(), this.getDirection());
	}
	
	//updates velocity vector
	public void move() {
		Point update = new Point(this.getDirection().getX(), this.getDirection().getY());
		update.scale(accel);
		this.getVelocity().add(update);
		this.setVelocity(this.getVelocity());
	}
	
	//slows the ship down
	public void slow() {
		if (this.getVelocity().length() > accel) {
			this.getVelocity().scale(1 - accel);
		}
		else
			this.setVelocity(new Point(0, 0));
	}
	
	public void applyGravity(Planet[] planets) {
		double g_xsum = 0.0;
		double g_ysum = 0.0;
		
		for(Planet p: planets) {
			double dx = p.getCentre().getX() - this.getCentre().getX();
			double dy = p.getCentre().getY() - this.getCentre().getY();
			double distsq = Math.pow(new Point(dx, dy).length(), 2);
			g_xsum += p.getGc() * dx / distsq;
			g_ysum += p.getGc() * dy / distsq;
		}
		this.getVelocity().add(new Point(g_xsum, g_ysum));
		this.setVelocity(this.getVelocity());
	}
	
	//updates ship location
	public void update() {
		this.getCentre().add(this.getVelocity());
		this.setHitbox(this.getCentre(), this.getDirection());
	}
	
	public void update(int x, int y) {
		this.getCentre().add(this.getVelocity());
		this.setCentre(new Point((1.0*x + this.getCentre().getX()) % x, (1.0*y + this.getCentre().getY()) % y));
		this.setHitbox(this.getCentre(), this.getDirection());
	}

	public Point getVelocity() {
		return velocity;
	}

	public void setVelocity(Point velocity) {
		if(velocity.length() > this.maxspd) {
			velocity = new Point(velocity.getX() * (maxspd/velocity.length()), velocity.getY() *  (maxspd/velocity.length()));
		}
		this.velocity = velocity;
	}
}
